--- a/luci-app-openvpn-server/root/etc/uci-defaults/openvpn
+++ b/luci-app-openvpn-server/root/etc/uci-defaults/openvpn
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+grep -qv "myvpn" /etc/config/openvpn && grep -q "myvpn" /etc/config/openvpn-opkg && cp -f /etc/config/openvpn-opkg /etc/config/openvpn
+
 openvpn_port="$(uci -q get openvpn.myvpn.port)"
 [ -z "$openvpn_port" ] && openvpn_port=1194

--- a/luci-app-openvpn-server/luasrc/controller/openvpn-server.lua
+++ b/luci-app-openvpn-server/luasrc/controller/openvpn-server.lua
@@ -6,13 +6,13 @@ function index()
 		return
 	end
 	
-	entry({"admin", "services", "openvpn-server"},firstchild(), _("OpenVPN Server"), 1).dependent = true
+	entry({"admin", "vpn", "openvpn-server"},firstchild(), _("OpenVPN Server"), 1).dependent = true
 
-	entry({"admin", "services", "openvpn-server", "general"}, cbi("openvpn-server/openvpn-server"), _("OpenVPN Server"), 1).leaf = true
-	entry({"admin", "services", "openvpn-server", "log"},form("openvpn-server/openvpn-server_run_log"), _("Running log"), 2).leaf = true
-	entry({"admin", "services", "openvpn-server", "passlog"},form("openvpn-server/openvpn-server_pass_log"), _("Login log"), 3).leaf = true
+	entry({"admin", "vpn", "openvpn-server", "general"}, cbi("openvpn-server/openvpn-server"), _("OpenVPN Server"), 1).leaf = true
+	entry({"admin", "vpn", "openvpn-server", "log"},form("openvpn-server/openvpn-server_run_log"), _("Running log"), 2).leaf = true
+	entry({"admin", "vpn", "openvpn-server", "passlog"},form("openvpn-server/openvpn-server_pass_log"), _("Login log"), 3).leaf = true
 
-	entry({"admin", "services", "openvpn-server","status"},call("act_status")).leaf=true
+	entry({"admin", "vpn", "openvpn-server","status"},call("act_status")).leaf=true
 end
 
 function act_status()

--- a/luci-app-openvpn-server/luasrc/model/cbi/openvpn-server/openvpn-server.lua
+++ b/luci-app-openvpn-server/luasrc/model/cbi/openvpn-server/openvpn-server.lua
@@ -166,7 +166,7 @@ if pid == "" then
         message = luci.util.exec("/etc/init.d/openvpn start 2>&1")
         luci.util.exec("sleep 2")
         luci.http.redirect(
-                luci.dispatcher.build_url("admin", "services", "openvpn-server") .. "?message=" .. message
+                luci.dispatcher.build_url("admin", "vpn", "openvpn-server") .. "?message=" .. message
         )
   end
 else
@@ -177,7 +177,7 @@ else
         luci.util.exec("/etc/init.d/openvpn stop")
         luci.util.exec("sleep 2")
         luci.http.redirect(
-                luci.dispatcher.build_url("admin", "services", "openvpn-server")
+                luci.dispatcher.build_url("admin", "vpn", "openvpn-server")
         )
   end
 end

--- a/luci-app-openvpn-server/luasrc/view/openvpn/openvpn_status.htm
+++ b/luci-app-openvpn-server/luasrc/view/openvpn/openvpn_status.htm
@@ -1,5 +1,5 @@
 <script type="text/javascript">//<![CDATA[
-XHR.poll(3, '<%=url([[admin]], [[services]], [[openvpn-server]], [[status]])%>', null,
+XHR.poll(3, '<%=url([[admin]], [[vpn]], [[openvpn-server]], [[status]])%>', null,
 	function(x, data) {
 		var tb = document.getElementById('openvpn_status');
 		if (data && tb) {

--- a/luci-app-openvpn-server/Makefile
+++ b/luci-app-openvpn-server/Makefile
@@ -10,7 +10,11 @@ LUCI_DEPENDS:=+openvpn-openssl +openvpn-easy-rsa
 LUCI_PKGARCH:=all
 PKG_NAME:=luci-app-openvpn-server
 PKG_VERSION:=2.0
-PKG_RELEASE:=21
+PKG_RELEASE:=22
+
+define Package/luci-app-openvpn-server/conffiles
+/etc/easy-rsa/vars.openvpn-server
+endef
 
 include $(TOPDIR)/feeds/luci/luci.mk
 

--- a/luci-app-openvpn-server/root/etc/easy-rsa/vars
+++ /dev/null
@@ -1,10 +0,0 @@
-set_var EASYRSA_OPENSSL	"openssl"
-set_var EASYRSA_KEY_SIZE	2048
-set_var EASYRSA_REQ_COUNTRY	"lunatickochiya"
-set_var EASYRSA_REQ_PROVINCE	"github"
-set_var EASYRSA_REQ_CITY	"github"
-set_var EASYRSA_REQ_ORG	"github"
-set_var EASYRSA_REQ_EMAIL	"lunatickochiya@github.com"
-set_var EASYRSA_REQ_OU	"github"
-set_var EASYRSA_CA_EXPIRE	3650
-set_var EASYRSA_CERT_EXPIRE	3650

--- /dev/null
+++ b/luci-app-openvpn-server/root/etc/easy-rsa/vars.openvpn-server
@@ -0,0 +1,10 @@
+set_var EASYRSA_OPENSSL	"openssl"
+set_var EASYRSA_KEY_SIZE	2048
+set_var EASYRSA_REQ_COUNTRY	"lunatickochiya"
+set_var EASYRSA_REQ_PROVINCE	"github"
+set_var EASYRSA_REQ_CITY	"github"
+set_var EASYRSA_REQ_ORG	"github"
+set_var EASYRSA_REQ_EMAIL	"lunatickochiya@github.com"
+set_var EASYRSA_REQ_OU	"github"
+set_var EASYRSA_CA_EXPIRE	3650
+set_var EASYRSA_CERT_EXPIRE	3650

--- a/luci-app-openvpn-server/root/etc/openvpncert.sh
+++ b/luci-app-openvpn-server/root/etc/openvpncert.sh
@@ -26,7 +26,7 @@ function rand_easy_rsa_vars() {
 		-e "s/^[[:space:]]*set_var[[:space:]]\+EASYRSA_REQ_ORG[[:space:]]\+\".*\"$/set_var EASYRSA_REQ_ORG\t\"$KEY_ORG\"/" \
 		-e "s/^[[:space:]]*set_var[[:space:]]\+EASYRSA_REQ_EMAIL[[:space:]]\+\".*\"$/set_var EASYRSA_REQ_EMAIL\t\"$KEY_EMAIL\"/" \
 		-e "s/^[[:space:]]*set_var[[:space:]]\+EASYRSA_REQ_OU[[:space:]]\+\".*\"$/set_var EASYRSA_REQ_OU\t\"$KEY_OU\"/" \
-		/etc/easy-rsa/vars
+		/etc/easy-rsa/vars.openvpn-server
 }
 
 rand_easy_rsa_vars
@@ -35,7 +35,7 @@ rand_easy_rsa_vars
 rm -rf /root/pki
 
 export EASYRSA_PKI="/etc/easy-rsa/pki"
-export EASYRSA_VARS_FILE="/etc/easy-rsa/vars"
+export EASYRSA_VARS_FILE="/etc/easy-rsa/vars.openvpn-server"
 export EASYRSA_CLI="easyrsa --batch"
 
 echo -en "yes\nyes\n" | $EASYRSA_CLI init-pki

--- a/luci-app-openvpn-server-client/root/etc/uci-defaults/openvpn
+++ b/luci-app-openvpn-server-client/root/etc/uci-defaults/openvpn
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+grep -qv "myvpn" /etc/config/openvpn && grep -q "myvpn" /etc/config/openvpn-opkg && cp -f /etc/config/openvpn-opkg /etc/config/openvpn
+
 openvpn_port="$(uci -q get openvpn.myvpn.port)"
 [ -z "$openvpn_port" ] && openvpn_port=1194

--- a/luci-app-openvpn-server-client/luasrc/controller/openvpn-server.lua
+++ b/luci-app-openvpn-server-client/luasrc/controller/openvpn-server.lua
@@ -18,13 +18,13 @@ function index()
 		return
 	end
 
-	entry({"admin", "services", "openvpn-server"},firstchild(), _("OpenVPN Server"), 1).dependent = true
+	entry({"admin", "vpn", "openvpn-server"},firstchild(), _("OpenVPN Server"), 1).dependent = true
 
-	entry({"admin", "services", "openvpn-server", "general"}, cbi("openvpn-server/openvpn-server"), _("OpenVPN Server"), 1).leaf = true
-	entry({"admin", "services", "openvpn-server", "client"},cbi("openvpn-server/openvpn-server_ovpn"), _("Client"), 2).leaf = true
-	entry({"admin", "services", "openvpn-server", "log"},form("openvpn-server/openvpn-server_run_log"), _("Running log"), 3).leaf = true
+	entry({"admin", "vpn", "openvpn-server", "general"}, cbi("openvpn-server/openvpn-server"), _("OpenVPN Server"), 1).leaf = true
+	entry({"admin", "vpn", "openvpn-server", "client"},cbi("openvpn-server/openvpn-server_ovpn"), _("Client"), 2).leaf = true
+	entry({"admin", "vpn", "openvpn-server", "log"},form("openvpn-server/openvpn-server_run_log"), _("Running log"), 3).leaf = true
 
-	entry({"admin", "services", "openvpn-server","status"},call("act_status")).leaf=true
+	entry({"admin", "vpn", "openvpn-server","status"},call("act_status")).leaf=true
 end
 
 function act_status()

--- a/luci-app-openvpn-server-client/luasrc/model/cbi/openvpn-server/openvpn-server.lua
+++ b/luci-app-openvpn-server-client/luasrc/model/cbi/openvpn-server/openvpn-server.lua
@@ -166,7 +166,7 @@ if pid == "" then
         message = luci.util.exec("/etc/init.d/openvpn start 2>&1")
         luci.util.exec("sleep 2")
         luci.http.redirect(
-                luci.dispatcher.build_url("admin", "services", "openvpn-server") .. "?message=" .. message
+                luci.dispatcher.build_url("admin", "vpn", "openvpn-server") .. "?message=" .. message
         )
   end
 else
@@ -177,7 +177,7 @@ else
         luci.util.exec("/etc/init.d/openvpn stop")
         luci.util.exec("sleep 2")
         luci.http.redirect(
-                luci.dispatcher.build_url("admin", "services", "openvpn-server")
+                luci.dispatcher.build_url("admin", "vpn", "openvpn-server")
         )
   end
 end

--- a/luci-app-openvpn-server-client/luasrc/view/openvpn/openvpn_status.htm
+++ b/luci-app-openvpn-server-client/luasrc/view/openvpn/openvpn_status.htm
@@ -1,5 +1,5 @@
 <script type="text/javascript">//<![CDATA[
-XHR.poll(3, '<%=url([[admin]], [[services]], [[openvpn-server]], [[status]])%>', null,
+XHR.poll(3, '<%=url([[admin]], [[vpn]], [[openvpn-server]], [[status]])%>', null,
 	function(x, data) {
 		var tb = document.getElementById('openvpn_status');
 		if (data && tb) {
